packages/endo/src/postinstall.js

```diff
@@ -6,14 +6,35 @@
 /* global process */
 
 import fs from 'fs';
+import { platform } from 'os';
+import { fileURLToPath } from 'url';
 
 const node = process.argv[0];
-const lockdown = new URL('./lockdown.cjs', import.meta.url).pathname;
-const mitm = new URL('../mitm/node', import.meta.url).pathname;
+const lockdown = fileURLToPath(new URL('./lockdown.cjs', import.meta.url));
+const mitm = fileURLToPath(
+  new URL(
+    `../mitm/node${platform() === 'win32' ? '.cmd' : ''}`,
+    import.meta.url,
+  ),
+);
 
-const script = `#!/bin/bash
+const nixScript = `#!/bin/bash
 set -ueo pipefail
 ${node} -r ${lockdown} "$@"`;
 
-fs.writeFileSync(mitm, script, 'utf-8');
+const win32Script = `
+@IF EXIST "%~dp0\\node.exe" (
+    "%~dp0\\node.exe" -r ${lockdown} -r %*
+) ELSE (
+    @SETLOCAL
+    @SET PATHEXT=%PATHEXT:;.JS;=;%
```

Comment:
Why is setting `PATHEXT` necessary?

Reply:
Have no idea, `cmd` is a horrifying language. I'm copying this from how `npm` generates cmd for bin files.